### ISSUE TITLE
Section on function passing.

### ISSUE THREAD

**Issue by indolering** [HaxeFoundation/HaxeManual#198]
The manual needs a section on function passing.  I'm not a serious programmer, so it may just be that I can't find it because I don't know what to search for.  If there is a section on this topic, then perhaps we could reword it to include "function passing" and link to it from [this SO post](http://stackoverflow.com/questions/14025739/passing-a-function-as-a-parameter-in-haxe).

I would write one myself, but (again) I'm not a serious programmer and I couldn't write something comprehensive.

**Comment by markknol** [comment]
Sounds good! you could start with a draft on the topic; we can help improve it before putting online.

**Comment by indolering** [comment]
@markknol How would the syntax work for a function that access an array of functions with different inputs but a single input?  Is Array<Dynamic> the best we can do?

**Comment by indolering** [comment]
The following throws an error if you don't declare the type for `var d`.

```Haxe
    private function diagnose(tests:Array<Dynamic>):Diagnosis{
        for(check in tests){
            var d:Diagnosis = check();
            if(!d.valid) return d;
        }

        return new Diagnosis();
    }

    public function test():Diagnosis {

        return diagnose([
            this.name.testName,
            this.rdata.testAddress,
            this.ttl.testTTL
        ]);
    }
```

**Comment by nadako** [comment]
why `Dynamic` when we have function types for that. just use `Void->Diagnosis`

**Comment by markknol** [comment]
Yes, that could be ` private function diagnose(tests:Array<Void->Diagnosis>):Diagnosis {`

I think we want to avoid as much `Dynamic` as possible in the manual, since most things can be nicely solved with types.

**Comment by indolering** [comment]
I think I need to just rewrite the section on function types : p

**Comment by Simn** [comment]
I don't think this needs any special content.
